servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -539,7 +589,62 @@ protected Publisher() {
+     * {@link Publisher}&lt;{@link R}&gt;.
+     * <p>
+     * The behavior is the same as {@link #flatMapMergeSingle(Function, int)} with the exception that if any
+     * {@link Single} returned by {@code mapper}, terminates with an error, the returned {@link Publisher} will not
+     * immediately terminate. Instead, it will wait for this {@link Publisher} and all {@link Single}s to terminate and
+     * then terminate the returned {@link Publisher} with all errors emitted by the {@link Single}s produced by the
+     * {@code mapper}.
+     * <p>
+     * This method is similar to {@link #map(Function)} but the result is asynchronous, and provides a data
+     * transformation in sequential programming similar to:
+     * <pre>{@code
+     *     Executor e = ...;
+     *     List<T> tResults = resultOfThisPublisher();
+     *     List<R> rResults = ...; // assume this is thread safe
+     *     List<Throwable> errors = ...;  // assume this is thread safe
+     *     CountDownLatch latch =  new CountDownLatch(tResults.size());
+     *     for (T t : tResults) {
+     *         // Note that flatMap process results in parallel.
+     *         e.execute(() -> {
+     *             try {
+     *                 R r = mapper.apply(t); // Asynchronous result is flatten into a value by this operator.
+     *                 rResults.add(r);
+     *             } catch (Throwable cause) {
+     *                 errors.add(cause);  // Asynchronous error is flatten into an error by this operator.
+     *             } finally {
+     *                 latch.countdown();
+     *             }
+     *         });
+     *     }
+     *     latch.await();
+     *     if (errors.isEmpty()) {
+     *         return rResults;
+     *     }
+     *     createAndThrowACompositeException(errors);
+     * }</pre>
+     *
+     * @param mapper {@link Function} to convert each item emitted by this {@link Publisher} into a {@link Single}.
+     * @param maxConcurrency Maximum active {@link Single}s at any time.
+     * Even if the number of items requested by a {@link Subscriber} is more than this number,
+     * this will never request more than this number at any point.
+     * @param maxDelayedErrorsHint The maximum amount of errors that will be queued. After this point exceptions maybe
+     * discarded to reduce memory consumption.
+     * @param <R> Type of items emitted by the returned {@link Publisher}.
+     * @return A new {@link Publisher} that emits all items emitted by each single produced by {@code mapper}.
+     *
+     * @see <a href="http://reactivex.io/documentation/operators/merge.html">ReactiveX merge operator.</a>
+     */
+    public final <R> Publisher<R> flatMapMergeSingleDelayError(
+            Function<? super T, ? extends Single<? extends R>> mapper, int maxConcurrency, int maxDelayedErrorsHint) {
+        return new PublisherFlatMapSingle<>(this, mapper, maxDelayedErrorsHint, maxConcurrency, executor);
```

Comment:
`maxDelayedErrorsHint == 0` has a special different meaning. Users may think that if they use `*DelayError` methods and pass `0` here, all exceptions will be delayed, but none of them will be accumulated in the `CompositeException`. However, they will start seeing exceptions asap.

Reply:
good point, `maxDelayedErrorsHint == 0` shouldn't be allowed here.